smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CodegenVisitor.java

```diff
@@ -317,6 +333,11 @@ public Void serviceShape(ServiceShape shape) {
                             paginationWriter));
         }
 
+        if (hasWaiter) {
+            writers.useFileWriter(WaiterGenerator.WAITABLE_INTERFACE_FILE, waiterWriter ->
```

Comment:
Correct, there is a small interface. This is what it looks like:
```
import { WeatherClient } from "../WeatherClient";
import { WaiterOptions } from "@aws-sdk/util-waiters";

export interface WeatherClientWaiter extends WaiterOptions {
  client: WeatherClient
}

```

Reply:
Like over all clients that takes a generic client type? 

None really, I like to try to avoid generics as much as possible - they tend to confuse people. But I can do that. I happen to think this is a lot more readable and it doesn't require the user to pass in a type generic to the exported waiter.

Reply:
Oh I see what you're saying. Yeah I can do that

Reply:
done: https://github.com/awslabs/smithy-typescript/pull/243/commits/ba77d90f56257e8c2c9bb8ee11852eaf906e4931

Reply:
```
waitForCitiesExist(params: WaiterOptions<WeatherClient>, ...){...} 
```